integrationtests/src/test/java/org/axonframework/integrationtests/testsuite/student/AbstractStudentTestSuite.java

```diff
@@ -197,31 +164,18 @@ protected <T, R> R sendCommand(T payload, Class<R> expectedResultType) {
         return commandGateway.sendAndWait(payload, expectedResultType);
     }
 
-    protected void verifyStudentName(String id, String name) {
-        UnitOfWork uow = unitOfWorkFactory.create();
-        uow.executeWithResult(context -> context.component(StateManager.class)
-                                                .repository(Student.class, String.class)
-                                                .load(id, context)
-                                                .thenAccept(student -> assertEquals(name,
-                                                                                    student.entity().getName())))
-           .join();
+    protected void studentEnrolledToCourse(String studentId, String courseId) {
+        storeEvent(StudentEnrolledEvent.class, new StudentEnrolledEvent(studentId, courseId));
     }
 
-    protected void verifyStudentEnrolledInCourse(String id, String courseId) {
+    protected <T> void storeEvent(Class<T> clazz, T payload) {
         UnitOfWork uow = unitOfWorkFactory.create();
-        uow.executeWithResult(context -> context.component(StateManager.class)
-                                                .repository(Student.class, String.class)
-                                                .load(id, context)
-                                                .thenAccept(student -> assertTrue(student.entity()
-                                                                                         .getCoursesEnrolled()
-                                                                                         .contains(courseId)))
-                                                .thenCompose(v -> context.component(StateManager.class)
-                                                                         .repository(Course.class,
-                                                                                     String.class)
-                                                                         .load(courseId, context))
-                                                .thenAccept(course -> assertTrue(course.entity()
-                                                                                       .getStudentsEnrolled()
-                                                                                       .contains(id))))
-           .join();
+        var eventMessage = new GenericEventMessage(
```

Comment:
Now that I have been diving into the test myself too: why use the `EventStore` directly here? We can simply publish events on the `EventGateway`, right? Or are you looking to get the tags in the right place?

Reply:
You're right. I've changed it to the EventGateway.